File: app/src/main/java/com/example/krystianwsul/organizator/sql/MySQLiteHelper.java
package com.example.krystianwsul.organizator.sql;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import com.example.krystianwsul.organizator.persistencemodel.CustomTimeRecord;
import com.example.krystianwsul.organizator.persistencemodel.DailyScheduleTimeRecord;
import com.example.krystianwsul.organizator.persistencemodel.InstanceRecord;
import com.example.krystianwsul.organizator.persistencemodel.ScheduleRecord;
import com.example.krystianwsul.organizator.persistencemodel.SingleScheduleDateTimeRecord;
import com.example.krystianwsul.organizator.persistencemodel.TaskHierarchyRecord;
import com.example.krystianwsul.organizator.persistencemodel.TaskRecord;
import com.example.krystianwsul.organizator.persistencemodel.WeeklyScheduleDayOfWeekTimeRecord;

public class MySQLiteHelper extends SQLiteOpenHelper {
    public static final String DATABASE_NAME = "tasks.db";
    public static final int DATABASE_VERSION = 1;

    public MySQLiteHelper(Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase sqLiteDatabase) {
        CustomTimeRecord.onCreate(sqLiteDatabase);
        DailyScheduleTimeRecord.onCreate(sqLiteDatabase);
        InstanceRecord.onCreate(sqLiteDatabase);
        ScheduleRecord.onCreate(sqLiteDatabase);
        SingleScheduleDateTimeRecord.onCreate(sqLiteDatabase);
        TaskHierarchyRecord.onCreate(sqLiteDatabase);
        TaskRecord.onCreate(sqLiteDatabase);
        WeeklyScheduleDayOfWeekTimeRecord.onCreate(sqLiteDatabase);
    }

    @Override
    public void onUpgrade(SQLiteDatabase sqLiteDatabase, int oldVersion, int newVersion) {
        CustomTimeRecord.onUpgrade(sqLiteDatabase, oldVersion, newVersion);
        DailyScheduleTimeRecord.onUpgrade(sqLiteDatabase, oldVersion, newVersion);
        InstanceRecord.onUpgrade(sqLiteDatabase, oldVersion, newVersion);
        ScheduleRecord.onUpgrade(sqLiteDatabase, oldVersion, newVersion);
        SingleScheduleDateTimeRecord.onUpgrade(sqLiteDatabase, oldVersion, newVersion);
        TaskHierarchyRecord.onUpgrade(sqLiteDatabase, oldVersion, newVersion);
        TaskRecord.onUpgrade(sqLiteDatabase, oldVersion, newVersion);
        WeeklyScheduleDayOfWeekTimeRecord.onUpgrade(sqLiteDatabase, oldVersion, newVersion);
    }
}

File: app/src/main/java/com/krystianwsul/checkme/persistencemodel/MySQLiteHelper.kt
package com.krystianwsul.checkme.persistencemodel

import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import com.krystianwsul.checkme.MyApplication

class MySQLiteHelper private constructor() : SQLiteOpenHelper(MyApplication.instance, DATABASE_NAME, null, DATABASE_VERSION) {

    companion object {

        private const val DATABASE_NAME = "tasks.db"
        private const val DATABASE_VERSION = 23

        val database by lazy { MySQLiteHelper().writableDatabase!! }
    }

    override fun onCreate(sqLiteDatabase: SQLiteDatabase) {
        InstanceShownRecord.onCreate(sqLiteDatabase)
    }

    override fun onUpgrade(sqLiteDatabase: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        sqLiteDatabase.beginTransaction()

        try {
            if (oldVersion < 23) {
                sqLiteDatabase.execSQL("DROP TABLE IF EXISTS customTimes")
                sqLiteDatabase.execSQL("DROP TABLE IF EXISTS dailySchedules")
                sqLiteDatabase.execSQL("DROP TABLE IF EXISTS instances")
                sqLiteDatabase.execSQL("DROP TABLE IF EXISTS monthlyDaySchedules")
                sqLiteDatabase.execSQL("DROP TABLE IF EXISTS monthlyWeekSchedules")
                sqLiteDatabase.execSQL("DROP TABLE IF EXISTS schedules")
                sqLiteDatabase.execSQL("DROP TABLE IF EXISTS singleSchedules")
                sqLiteDatabase.execSQL("DROP TABLE IF EXISTS taskHierarchies")
                sqLiteDatabase.execSQL("DROP TABLE IF EXISTS tasks")
                sqLiteDatabase.execSQL("DROP TABLE IF EXISTS weeklySchedules")
            }

            sqLiteDatabase.setTransactionSuccessful()
        } finally {
            sqLiteDatabase.endTransaction()
        }
    }
}
